PubSub.DataAccess/SubscriptionRepository.cs

```diff
@@ -0,0 +1,53 @@
+﻿using CSharpFunctionalExtensions;
+using Microsoft.EntityFrameworkCore;
+using PubSub.Core.Models;
+
+namespace PubSub.DataAccess;
+
+public class SubscriptionRepository : ISubscriptionRepository
+{
+    private readonly PubSubContext _context;
+    public SubscriptionRepository(PubSubContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<IEnumerable<Subscription>> GetAllByUserId(string userId)
+    {
+        return await _context.Subscriptions.Where(x => x.OwnerId == userId).ToListAsync();
+    }
+
+    public async Task<IEnumerable<Subscription>> GetByTopic(string topic)
+    {
+        return await SubscriptionsByTopic(topic).ToListAsync();
+    }
+
+    public async Task<bool> Exists(string topic, string url)
+    {
+        return await SubscriptionsByTopic(topic).Where(x => x.Callback == url).AnyAsync();
```

Comment:
Where can be skipped
```suggestion
        return await SubscriptionsByTopic(topic).AnyAsync(x => x.Callback == url);
```